Zend/zend_closures.c

```diff
@@ -235,6 +235,105 @@ ZEND_METHOD(Closure, bind)
 }
 /* }}} */
 
+static void zend_closure_call_magic(INTERNAL_FUNCTION_PARAMETERS) {
+	zend_fcall_info fci;
+	zend_fcall_info_cache fcc;
+
+	memset(&fci, 0, sizeof(zend_fcall_info));
+	memset(&fci, 0, sizeof(zend_fcall_info_cache));
+
+	fci.size = sizeof(zend_fcall_info);
+	fci.retval = return_value;
+
+	fcc.initialized = 1;
+	fcc.function_handler = (zend_function *) EX(func)->common.arg_info;
+	fci.params = (zval*) emalloc(sizeof(zval) * 2);
+	fci.param_count = 2;
+	ZVAL_STR(&fci.params[0], EX(func)->common.function_name);
+	array_init(&fci.params[1]);
+	zend_copy_parameters_array(ZEND_NUM_ARGS(), &fci.params[1]);
+
+	fci.object = Z_OBJ(EX(This));
+	fcc.object = Z_OBJ(EX(This));
+	fcc.calling_scope = zend_get_executed_scope();
+
+	zend_call_function(&fci, &fcc);
+
+	zval_ptr_dtor(&fci.params[0]);
+	zval_ptr_dtor(&fci.params[1]);
+	efree(fci.params);
+}
```

Comment:
Also generally, whatever issue this code is solving, does the same issue exist for ReflectionMethod::getClosure() and if not, how did they solve it?